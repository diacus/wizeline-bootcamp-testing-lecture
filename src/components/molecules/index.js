export { default as InputNumber } from './InputNumber/InputNumber';
export { default as DaysSince } from './DaysSince/DaysSince';
export { default as UserInputField } from './UserInputField/UserInputField';
