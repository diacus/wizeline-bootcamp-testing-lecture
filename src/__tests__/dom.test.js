import React from 'react';
import ReactDOM from 'react-dom';

import { getByPlaceholderText, getByText, queryByTestId, waitFor } from '@testing-library/dom';
import '@testing-library/jest-dom';

import { UserInputField } from '../components/molecules';



function getExampleDOM() {
  const div = document.createElement('div');

  ReactDOM.render(<UserInputField />, div);

  return div;
}

test('examples of some things', async () => {
  const anyUser = 'Cosme Fulanito'
  const container = getExampleDOM();

  const input = getByPlaceholderText(container, 'Nombre');
  input.value = anyUser;

  getByText(container, 'Click me!').click();

  await waitFor(() => {
    const expectedText = 'Hola mi nombre es Cosme Fulanito';
    expect(queryByTestId(container, 'user-field-output')).toBeTruthy();
  });

  // jest snapshots work great with regular DOM nodes!
  expect(container).toMatchSnapshot()
})
