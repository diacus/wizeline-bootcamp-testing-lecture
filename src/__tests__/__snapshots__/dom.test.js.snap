// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples of some things 1`] = `
<div>
  <div
    class="sc-bdvvtL ivCekc"
  >
    <div
      class="sc-bdvvtL sc-gsDKAQ dOtxPc eEPFrg"
      display="flex"
    >
      <input
        class="sc-bdvvtL sc-gsDKAQ sc-gKclnd dOtxPc eEPFrg yyZpN"
        display="flex"
        id="user-input"
        placeholder="Nombre"
      />
    </div>
    <div
      class="sc-bdvvtL sc-gsDKAQ dOtxPc eEPFrg"
      display="flex"
    >
      <button
        class="sc-bdvvtL sc-hKwDye ivCekc hKqRVc"
      >
        Click me!
      </button>
    </div>
    <div
      class="sc-bdvvtL sc-gsDKAQ dOtxPc eEPFrg"
      display="flex"
    >
      <div
        class="sc-bdvvtL sc-iqseJM kLzALT hkpbPB"
        color="black"
        data-testid="user-field-output"
      >
        Hola mi nombre es 
        Cosme Fulanito
      </div>
    </div>
  </div>
</div>
`;
